### PR TITLE
58007 Service Enumeration, More Stubs

### PR DESCRIPTION
**Pull request by imaginebeingatcomputers**
- Added service enumeration via xlivebase message 58007
- success stub 50079

### CODE REVIEW

File: src/xenia/kernel/XLiveAPI.cpp [hunk 1]
```diff
@@ -1089,6 +1089,38 @@ std::vector<XLiveAPI::XTitleServer> XLiveAPI::GetServers() {
...
 }
 
+XLiveAPI::XONLINE_SERVICE_INFO XLiveAPI::GetServiceInfoById(
+    xe::be<uint32_t> serviceId) {
+  std::string endpoint = fmt::format("title/{:x}/services/{:08x}",
+                                     kernel_state()->title_id(), serviceId);
+
+  memory chunk = Get(endpoint);
+
+  XLiveAPI::XONLINE_SERVICE_INFO service{};
```
Comment (AdrianCassar) [inline src/xenia/kernel/XLiveAPI.cpp]:
```suggestion
  XONLINE_SERVICE_INFO service{};
```

File: src/xenia/kernel/XLiveAPI.cpp [hunk 2]
```diff
@@ -1089,6 +1089,38 @@ std::vector<XLiveAPI::XTitleServer> XLiveAPI::GetServers() {
...
+    assert_always();
+
+    return service;
+  }
+
+  Document doc;
+  doc.Parse(chunk.response);
+
+  for (const auto& service_info : doc.GetArray()) {
+    inet_pton(AF_INET, service_info["address"].GetString(), &(service.ip));
```
Comment (AdrianCassar) [inline src/xenia/kernel/XLiveAPI.cpp]:
```suggestion
    inet_pton(AF_INET, service_info["address"].GetString(), &service.ip);
```

File: src/xenia/kernel/XLiveAPI.cpp [hunk 3]
```diff
@@ -1089,6 +1089,38 @@ std::vector<XLiveAPI::XTitleServer> XLiveAPI::GetServers() {
...
+
+    XELOGD("GetServiceById IP: {}", service.ip.s_addr);
```
Comment (AdrianCassar) [inline src/xenia/kernel/XLiveAPI.cpp]:
```suggestion
    XELOGD("GetServiceById IP: {}", service_info["address"].GetString());
```

File: src/xenia/kernel/XLiveAPI.cpp (hunk 1 above)
Comment (imaginebeingatcomputers) [reply src/xenia/kernel/XLiveAPI.cpp]: We still seem to be scoping XLiveAPI within this module.

example:
https://github.com/AdrianCassar/xenia-canary/pull/3/files#diff-592a9d5dec9134ecdd51e2f042864c3b634c4f4fc2c1f534c9ca271a309db2b0L1059

**Pull request merged**